invenio_communities/views/decorators.py

```diff
@@ -29,6 +29,7 @@ def view(**kwargs):
                 id_=pid_value, identity=g.identity
             )
             kwargs["community"] = community
+            request.community = community.to_dict()
```

Comment:
We normally use the `g` context, and not the `request` context. From the Flask doc:
- [g](https://flask.palletsprojects.com/en/2.3.x/api/#flask.g): `This is a good place to store resources during a request.`
- [request](https://flask.palletsprojects.com/en/2.3.x/api/#flask.request): `To access incoming request data, you can use the global request object. Flask parses incoming request data for you and gives you access to it through that global object.`

Minor: `community.to_dict()` is repeated below, it could be set once, and re-used.

Moderate: can you add a comment to explain why? It won't be easy to understand why this was done, and it might break quite easily.